support/mesos-mini/mesos/agent_environment

```diff
@@ -3,7 +3,7 @@ MESOS_HOSTNAME=localhost
 MESOS_WORK_DIR=/var/lib/mesos/agent
 MESOS_MASTER=127.0.0.1:5050
 MESOS_CONTAINERIZERS=mesos,docker
-MESOS_ISOLATION=filesystem/linux,network/cni,cgroups/cpu,cgroups/mem,cgroups/blkio,cgroups/devices,disk/du,docker/runtime,volume/sandbox_path,volume/host_path,posix/rlimits,namespaces/pid,linux/capabilities
```

Comment:
No problem. :-)

```
Mar 31 09:57:55 andreas-pc mesos-agent[40]: Reached unreachable statement at linux/capabilities.cpp:497
Mar 31 09:57:55 andreas-pc mesos-agent[40]: *** Aborted at 1648720675 (unix time) try "date -d @1648720675" if you are using GNU date ***
Mar 31 09:57:55 andreas-pc mesos-agent[40]: PC: @     0x7f2e97e6f387 __GI_raise
Mar 31 09:57:55 andreas-pc mesos-agent[40]: *** SIGABRT (@0x16) received by PID 22 (TID 0x7f2e9db92a00) from PID 22; stack trace: ***
Mar 31 09:57:55 andreas-pc mesos-agent[40]:     @     0x7f2e98733630 (unknown)
Mar 31 09:57:55 andreas-pc mesos-agent[40]:     @     0x7f2e97e6f387 __GI_raise
Mar 31 09:57:55 andreas-pc mesos-agent[40]:     @     0x7f2e97e70a78 __GI_abort
Mar 31 09:57:55 andreas-pc mesos-agent[40]:     @     0x7f2e9ae164d7 Unreachable()
Mar 31 09:57:55 andreas-pc mesos-agent[40]:     @     0x7f2e9b81dba1 mesos::internal::capabilities::operator<<()
Mar 31 09:57:55 andreas-pc mesos-agent[40]:     @     0x7f2e9b8217ac stringify<>()
Mar 31 09:57:55 andreas-pc mesos-agent[40]:     @     0x7f2e9b820076 mesos::internal::capabilities::Capabilities::create()
Mar 31 09:57:55 andreas-pc mesos-agent[40]:     @     0x7f2e9b908c2d mesos::internal::slave::LinuxCapabilitiesIsolatorProcess::create()
Mar 31 09:57:55 andreas-pc mesos-agent[40]:     @     0x7f2e9b42f9ec std::_Function_handler<>::_M_invoke()
Mar 31 09:57:55 andreas-pc mesos-agent[40]:     @     0x7f2e9b41c9f2 mesos::internal::slave::MesosContainerizer::create()
Mar 31 09:57:55 andreas-pc mesos-agent[40]:     @     0x7f2e9b399696 mesos::internal::slave::Containerizer::create()
Mar 31 09:57:55 andreas-pc mesos-agent[40]:     @     0x55d68541b10a (unknown)
Mar 31 09:57:55 andreas-pc mesos-agent[40]:     @     0x7f2e97e5b555 __libc_start_main
Mar 31 09:57:55 andreas-pc mesos-agent[40]:     @     0x55d68541e221 (unknown)
Mar 31 09:57:56 andreas-pc systemd[1]: mesos-slave.service: main process exited, code=dumped, status=6/ABRT
Mar 31 09:57:56 andreas-pc systemd[1]: Unit mesos-slave.service entered failed state.
Mar 31 09:57:56 andreas-pc systemd[1]: mesos-slave.service failed.
```

Reply:
Hmm, it seemed to run into https://github.com/apache/mesos/blob/1.11.0/src/linux/capabilities.cpp#L497. Can you please let me know how you ran Mesos agent? Like the OS version, in a Docker container? I'd suggest to figure out why it ran into that unreachable statement first.

Reply:
Hi @qianzhangxa , I found out, I have the same issue under Debian bullseye (outside of docker). Under Ubuntu focal I does not have this issue. :man_shrugging:

Reply:
It match to this one: https://github.com/apache/mesos/pull/377

Reply:
Ok, these capabilities are missing:

-        CAP_BPF (since Linux 5.8)
-        CAP_CHECKPOINT_RESTORE (since Linux 5.9)
-        CAP_PERFMON (since Linux 5.8)

Reply:
@andreaspeters So I think there will be no such issue if you use the latest code which already contains those missing capabilities (https://github.com/apache/mesos/blob/master/include/mesos/mesos.proto), right?

Reply:
Right! Well, maybe we should plan a new release or just a patched version therefore all the peoples who want try out mesos on a new Linux does not run into this issue. What do u think?

Reply:
Yeah, a patch release sounds good to me.

Reply:
:+1:

Reply:
How is it going with the patch release?

Reply:
Sorry, I do not have the free cycle to do a patch release 😞

Reply:
:-( What can I do to support you?